Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -3861,6 +3861,15 @@ void WebPage::runJavaScript(WebFrame* frame, RunJavaScriptParameters&& parameter
     }
 #endif
 
+    bool shouldAllowUserInteraction = [&] {
+#if PLATFORM(COCOA)
+        if (linkedOnOrAfterSDKWithBehavior(SDKAlignedBehavior::ProgrammaticFocusDuringUserScriptShowsInputViews))
+            return true;
+#endif
+        return m_userIsInteracting;
+    }();
+
+    SetForScope userIsInteractingChange { m_userIsInteracting, shouldAllowUserInteraction };
```

Comment:
do we also want to check `parameters.forceUserGesture`?  i think that `-[WKWebView _evaluateJavaScriptWithoutUserGesture:completionHandler:]` would probably not want to set the user interaction flag 😅

(part of me wants to ask you to move this to `ScriptController::executeScriptInWorld` since that's where the `UserGestureIndicator` is, but i understand that would involve lotsa `PageClient` plumbing so it's probably fine here)